### ISSUE TITLE
Dependencies

### ISSUE THREAD

**Issue by ghost** [BobRay/MyComponent#30]
Does MyComponent deal with 2.4's dependencies support?

**Comment by BobRay** [comment]
No, unfortunately not at all, though it's *very* easy to have it package up other extras and install them.

**Comment by Gibbs** [comment]
How do you package up other extras?

In regard to dependencies I've found them quite difficult to get working. Some dependencies get listed as un-installed even when they are, and some fail to download due to bad signatures. Anyway you can add/hack them to your ```build.transport.php``` file under ```assets/mycomponents/yourcomponent/_build/```. Example;

```php
/* Next-to-last step - pack in the license file, readme.txt, changelog,
 * and setup options 
 */
$attr = array(
    'license' => file_get_contents($sources['docs'] . 'license.txt'),
    'readme' => file_get_contents($sources['docs'] . 'readme.txt'),
    'changelog' => file_get_contents($sources['docs'] . 'changelog.txt'),

    // This adds MODX 2.4+ dependencies
    'requires' => array(
      'pthumb'                => '>=2.3.3',
      'migx'                  => '>=2.9.6',
      'pdotools'              => '>=2.1.20',
    ),
);
```

**Comment by sottwell** [comment]
I've never been able to get dependencies to work, and most of the extras I've installed that used them also didn't work.

**Comment by Gibbs** [comment]
I'm not 100% but I think it depends on the extras included. I can never get, for example, "tinymcerichtexteditor" to work. No idea why.

**Comment by BobRay** [comment]
I'm not sure I understand what you folks are saying. MyComponent has its own dependency management via subpackages. Are you saying that the native MyComponent dependency system doesn't work, or that the MODX system of adding the array to the build.transport.php file doesn't work in general, or that using the MODX system doesn't work under MyComponent?

**Comment by sottwell** [comment]
Sorry, we got sidetracked on the 2.4 dependencies business. Subpackages is what Gibbs was asking about originally.

**Comment by BobRay** [comment]
So you're saying that Gibbs' method of adding the array to build.transport.php works better than the native MyComponent subpackages system?

**Comment by sottwell** [comment]
Well, he appears to have some success with the dependencies, but I never have, at least not up to this time.

**Comment by BobRay** [comment]
MyComponent has been modified to use the new MODX dependency system
